api/src/discord/listeners/interactionCreate.ts

```diff
@@ -0,0 +1,234 @@
+                    }
+                });
+
+                interface Message {
+                    channel: string,
+                    content: string,
+                    author: string,
+                    timestamp: string
+                }
+
+                // put the archive in the archive channel of the ctf in the description
+                const niceMessages: string[] = allMessages.map((messages) => {
+                    let channelName = "";
+                    let niceMessage = "";
+
+                    messages = messages.reverse();
+
+                    if (messages.length > 0) {
+                        channelName = messages[0].channel;
+                        niceMessage += `## ${channelName}\n`;
+
+                        messages.forEach((message: Message) => {
+
+                            if (channelName != message.channel) {
+                                channelName = message.channel;
+                                niceMessage = `## ${channelName}\n`;
+                            }
+
+                            const timestamp = new Date(message.timestamp).toLocaleString();
+
+                            const formattedMessage = `[${timestamp}] ${message.author}: ${message.content}`;
+                            niceMessage += formattedMessage + "\n";
+
+
+                        });
+                    }
+
+                    return niceMessage;
+                });
+
+
+                const padUrl = await createPad(`${ctfName} archive`, niceMessages.join('\n'));
+                const ctfId = Number(await getCtfIdFromDatabase(ctfName));
+
+
+                createTask(`${ctfName} archive`, `Archive of ${ctfName}`, "archive", "", padUrl, ctfId)
+                // remove message
+                interaction.deleteReply();
+
+
+            }
+        }
+
+        if (interaction.isCommand() || interaction.isContextMenuCommand()) {
+            await handleSlashCommand(client, interaction);
+        }
+
+
+    });
+};
+
+
+async function fetchAllMessages(channel: TextBasedChannel): Promise<any> {
+    const messages = await channel.messages.fetch({limit: 100});
+
+    const messagesCollection: any[] = [];
+
+
+    messages.forEach((message: Message) => {
+
+        if (message.author.bot) return;
+        if (message.content.startsWith("/")) return;
+
+
+        const channel = message.channel as TextChannel
+
+        const channelName = channel.name
+        const timestamp = message.createdTimestamp
+        const author = message.author.username
+
+        let content = ""
+
+        if (message.attachments.size > 0) {
+            message.attachments.forEach((attachment) => {
+                message.content += attachment.url + " "
+            });
+        }
+
+        content += message.content
+
+        const messageObject = {
+            channel: channelName,
+            content: content,
+            author: author,
+            timestamp: timestamp
+        };
```

Comment:
Can we reuse the message interface you defined here above? Maybe a bit nicer for code quality